pkg/platform/openstack/kuryr_bootstrap.go

```diff
@@ -189,6 +194,57 @@ func getSavedAnnotation(kubeClient client.Client, annotation string) (string, er
 	return cm.Annotations[annotation], nil
 }
 
+func getMasterMachines(kubeClient client.Client) (*machineapi.MachineList, error) {
+	machineList := &machineapi.MachineList{}
+	listOptions := []client.ListOption{
+		client.InNamespace(machinesNamespace),
+		client.MatchingLabels{masterMachineLabel: "master"},
+	}
+	err := kubeClient.List(context.TODO(), machineList, listOptions...)
+	if err != nil {
+		return nil, err
+	}
+	return machineList, nil
+}
+
+func getWorkersSubnetFromMasters(client *gophercloud.ServiceClient, kubeClient client.Client, clusterID string) (subnets.Subnet, error) {
+	empty := subnets.Subnet{}
+	machines, err := getMasterMachines(kubeClient)
+	if err != nil {
+		return empty, err
+	}
+	for _, machine := range machines.Items {
```

Comment:
what happens if `len(machines.Items) == 0`?

Reply:
would that ever be reached?

Reply:
This would be considered a weird situation as Machines for master nodes are created by the installer. With 4.7 we do not have another way of determining the main worker subnet, so it'd fail on `findOpenStackSubnetByNetworkTag()` execution by default. Interestingly in envs with BYON upgraded from 4.6 to 4.7 the `findOpenStackSubnetByNetworkTag()` would successfully find the network - thanks to guarantee that it's tagged in Neutron. This is gone in 4.7 due to users requests (some doesn't have permissions to tag networks).

Anyway if there are no Machines created for masters then there's not much we can do and I'd be happy that CNO fails bootstrapping process here. It will be retried if somehow the Machines come back.